services/repository/contributors_graph.go

```diff
@@ -150,6 +151,30 @@ func getExtendedCommitStats(repo *git.Repository, revision string /*, limit int
 				authorEmail := strings.TrimSpace(scanner.Text())
 				scanner.Scan()
 				date := strings.TrimSpace(scanner.Text())
+
+				var coAuthors []*api.CommitUser
+				emailSet := map[string]bool{}
+				for scanner.Scan() {
+					line := scanner.Text()
+					if line == "" {
+						// There should be an empty line before we read the commit stats line.
+						break
+					}
+					coAuthorName, coAuthorEmail, err := util.ParseCommitTrailerValueWithAuthor(line)
+					if err != nil {
+						continue
+					}
+					if _, exists := emailSet[coAuthorEmail]; exists {
```

Comment:
We have a hash map here to account trailers with duplicate emails. Is it safe that this hash map is unbounded? If so, should we just ignore all the co-author data for this commit if they reach some certain bound?